lib/models/Review.js

```diff
@@ -0,0 +1,52 @@
+const pool = require('../utils/pool');
+
+class Review {
+  id;
+  user_id;
+  restaurant_id;
+  stars;
+  detail;
+
+  constructor(row) {
+    this.id = row.id;
+    this.user_id = row.user_id;
+    this.restaurant_id = row.restaurant_id;
+    this.stars = row.stars;
+    this.detail = row.detail;
+  }
+
+  static async getById(id) {
+    const { rows } = await pool.query(
+      `
+    SELECT * FROM reviews 
+    WHERE id = $1`,
+      [id]
+    );
+    if (rows.length === 0) {
+      return null;
+    }
+    return new Review(rows[0]);
+  }
+
+  static async insert({ userId, restaurantId, stars, detail }) {
+    const { rows } = await pool.query(
+      'INSERT INTO reviews (user_id, restaurant_id, stars, detail) VALUES ($1, $2, $3, $4) RETURNING *',
+      [userId, restaurantId, stars, detail]
+    );
+    return new Review(rows[0]);
+  }
+
+  static async delete(id) {
+    const { rows } = await pool.query(
+      `
+      DELETE from reviews
+      WHERE id = $1
+      RETURNING *
+    `,
+      [id]
+    );
```

Comment:
what happens if there is no review with the given id? I believe either the middleware will say "Access Denied" or, ```rows``` will be undefined, and then when you try and make a new ```Review``` on line 48, it will throw an error because ```rows[0]``` will be undefined. I don't think you tested for deleting a review that doesn't exist — correct me if I missed it